src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployerBase.cs

```diff
@@ -26,49 +28,62 @@ public IISDeployerBase(IISDeploymentParameters deploymentParameters, ILoggerFact
 
         public void RunWebConfigActions(string contentRoot)
         {
-            if (IISDeploymentParameters == null)
-            {
-                return;
-            }
-
             if (!DeploymentParameters.PublishApplicationBeforeDeployment)
             {
                 throw new InvalidOperationException("Cannot modify web.config file if no published output.");
             }
 
             var path = Path.Combine(DeploymentParameters.PublishedApplicationRootPath, "web.config");
             var webconfig = XDocument.Load(path);
-            var xElement = webconfig.Descendants("system.webServer").Single();
 
             foreach (var action in DefaultWebConfigActions)
             {
-                action.Invoke(xElement, contentRoot);
+                action.Invoke(webconfig.Root, contentRoot);
             }
 
-            foreach (var action in IISDeploymentParameters.WebConfigActionList)
+            if (IISDeploymentParameters != null)
             {
-                action.Invoke(xElement, contentRoot);
+                foreach (var action in IISDeploymentParameters.WebConfigActionList)
```

Comment:
Nit: `IISDeploymentParameters?.WebConfigActionList`

Reply:
You can't `foreach` null